crates/common/mqtt_channel/src/config.rs

```diff
@@ -120,6 +128,24 @@ impl Config {
         }
     }
 
+    /// Set the last will message, this will be published when the mqtt connection gets closed.
+    pub fn with_last_will_message(
+        self,
+        topic: impl Into<String>,
+        payload: impl Into<Vec<u8>>,
+    ) -> Self {
```

Comment:
Passing a complete message gives the caller an opportunity to choose the QoS and retain flag.

```suggestion
    pub fn with_last_will_message(
        self,
        message: Message
    ) -> Self {
```

Reply:
fixed